quickfixj-core/src/main/java/quickfix/mina/initiator/AbstractSocketInitiator.java

```diff
@@ -69,9 +76,27 @@ protected AbstractSocketInitiator(Application application,
 
     protected AbstractSocketInitiator(SessionSettings settings, SessionFactory sessionFactory)
             throws ConfigError {
+        this(settings, sessionFactory, 0);
+    }
+
+    protected AbstractSocketInitiator(Application application,
+            MessageStoreFactory messageStoreFactory, SessionSettings settings,
+            LogFactory logFactory, MessageFactory messageFactory, int numReconnectThreads) throws ConfigError {
+        this(settings, new DefaultSessionFactory(application, messageStoreFactory, logFactory,
+                messageFactory), numReconnectThreads);
+    }
+
+    protected AbstractSocketInitiator(SessionSettings settings, SessionFactory sessionFactory, int numReconnectThreads)
+            throws ConfigError {
         super(settings, sessionFactory);
         IoBuffer.setAllocator(new SimpleBufferAllocator());
         IoBuffer.setUseDirectBuffer(false);
+        if (numReconnectThreads > 0) {
+            scheduledReconnectExecutor = Executors.newScheduledThreadPool(numReconnectThreads, new QFScheduledReconnectThreadFactory());
+            ((ThreadPoolExecutor) scheduledReconnectExecutor).setMaximumPoolSize(numReconnectThreads);
```

Comment:
Setting `maximumPoolSize` to `initialPoolSize` ensures that `numReconnectThreads` threads are available to service tasks. Otherwise new threads only get created when the task queue is full.